src/otx/core/model/entity/classification.py

```diff
@@ -34,6 +36,33 @@ class OTXMulticlassClsModel(
 ):
     """Base class for the classification models used in OTX."""
 
+    @abstractmethod
+    def head_forward_fn(self, x):
+        """Defines neck (if available) and head forward function."""
+
+    @property
+    @abstractmethod
+    def has_gap(self):
+        """Defines if GAP is used right after backbone."""
+
+    def register_explain_hook(self, backbone: torch.nn.Module) -> None:
+        """Register explain hook at the model backbone output."""
+        self.explain_hook = ReciproCAMHook(
+            self.head_forward_fn,
+            num_classes=self.num_classes,
+            optimize_gap=self.has_gap,
+        )
+        self.explain_hook.handle = backbone.register_forward_hook(self.explain_hook.recording_forward)
```

Comment:
Can we wrap all the things for XAI algo inside of itself to clean up?

```suggestion
class ReciproCAMHook:
    def __init__(self, head_forward_fn, backbone, num_classes, optimize_gap):
        self.handle = backbone.register_forward_hook(self.recording_forward)

...

        self.explain_hook = ReciproCAMHook(
            head_forward_fn=self.head_forward_fn,
            backbone=backbone,
            num_classes=self.num_classes,
            optimize_gap=self.has_gap,
        )
```

Reply:
I created a class method that does that.